rest/tests/base.py

```diff
@@ -388,3 +388,80 @@ def delete_term(self, repo_slug, vocab_slug, term_slug,
             )
         )
         self.assertEqual(expected_status, resp.status_code)
+
+    def build_members_url(self, urlfor, repo_slug,
+                          username=None, group_type=None):
+        """
+        Helper function to build a url for members.
+        """
+        self.assertTrue(urlfor in ['base', 'users', 'groups'])
+        base_url = '{api_base}repositories/{repo_slug}/members/'
+        users_url = '{base_url}users/{username}/groups/'
+        groups_url = '{base_url}groups/{group_type}/users/'
+        extra_url = '{base_url}{extra}/'
+        url_for_repo = base_url.format(
+            api_base=API_BASE,
+            repo_slug=repo_slug
+        )
+        if urlfor == 'base':
+            return url_for_repo
+        elif urlfor == 'users':
+            url = users_url.format(base_url=url_for_repo, username=username)
+            if group_type is not None:
+                url = extra_url.format(base_url=url, extra=group_type)
+            return url
+        elif urlfor == 'groups':
+            url = groups_url.format(
+                base_url=url_for_repo,
+                group_type=group_type
+            )
+            if username is not None:
+                url = extra_url.format(base_url=url, extra=username)
+            return url
+
+    def get_members(self, urlfor, repo_slug,
+                    username=None, group_type=None,
+                    expected_status=HTTP_200_OK,
+                    skip_options_head_test=False):
+        """Get members"""
+        url = self.build_members_url(urlfor, repo_slug, username, group_type)
+        if not skip_options_head_test:
+            self.assert_options_head(url, expected_status=expected_status)
+        resp = self.client.get(url)
+        self.assertEqual(expected_status, resp.status_code)
+        if expected_status == HTTP_200_OK:
+            return as_json(resp)
+
+    def create_member(self, urlfor, repo_slug, mem_dict,
+                      username=None, group_type=None, reverse_str=None,
+                      expected_status=HTTP_201_CREATED, skip_assert=False):
+        """
+        Create member (meaning assigning an user to a repository group)
+        """
+        url = self.build_members_url(urlfor, repo_slug, username, group_type)
+        resp = self.client.post(url, mem_dict)
+        self.assertEqual(expected_status, resp.status_code)
+        if resp.status_code == HTTP_201_CREATED:
+            result_dict = as_json(resp)
+            if not skip_assert:
+                for key, value in mem_dict.items():
+                    self.assertEqual(value, result_dict[key])
+            if reverse_str is not None:
+                self.assertIn(reverse(
+                    reverse_str, kwargs={
+                        'username': username or mem_dict['username'],
```

Comment:
Interesting. I was sure this was invalid; I thought it needed to be `username if username is None else mem_dict['username']`. So, I learned something today. This syntax is nice, but confusing because it _looks_ like it should return a boolean.